cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlUnitTestUtils.java

```diff
@@ -74,18 +69,29 @@ public static AggregatedMetricValues getAggregatedMetricValues(double cpuUsage,
                                                                  double networkInBoundUsage,
                                                                  double networkOutBoundUsage,
                                                                  double diskUsage) {
-    double[] values = new double[Resource.cachedValues().size()];
-    values[KafkaMetricDef.resourceToMetricId(CPU)] = cpuUsage;
-    values[KafkaMetricDef.resourceToMetricId(NW_IN)] = networkInBoundUsage;
-    values[KafkaMetricDef.resourceToMetricId(NW_OUT)] = networkOutBoundUsage;
-    values[KafkaMetricDef.resourceToMetricId(DISK)] = diskUsage;
     AggregatedMetricValues aggregateMetricValues = new AggregatedMetricValues();
-    for (Resource r : Resource.cachedValues()) {
-      int metricId = KafkaMetricDef.resourceToMetricId(r);
+    setValueForResource(aggregateMetricValues, Resource.CPU, cpuUsage);
+    setValueForResource(aggregateMetricValues, Resource.NW_IN, networkInBoundUsage);
+    setValueForResource(aggregateMetricValues, Resource.NW_OUT, networkOutBoundUsage);
+    setValueForResource(aggregateMetricValues, Resource.DISK, diskUsage);
+    return aggregateMetricValues;
+  }
+
+  /**
+   * Set the utilization values of all metrics for a resource in the given AggregatedMetricValues.
+   * The first metric has the full resource utilization value, all the rest of the metrics has 0.
+   */
+  public static void setValueForResource(AggregatedMetricValues aggregatedMetricValues,
```

Comment:
Why do we set this only for the first metric? Is this because we decided to pick the first metric as the representative for the rest? Can we add clarification to the JavaDoc?

Reply:
This is to avoid massive changes to the class constants. We can change that in a separate patch.